### DISCUSSION TITLE
Gluon SceneBuilder - Installing without OpenJDK/Changing of default JDK

### DISCUSSION THREAD

**Discussion by tmcke01** [gluonhq/scenebuilder#644] [Q&A]
Creating this post as I reached out to Gluon Support and they promptly passed me on to this forum without answering my query. 

I have a technical question surrounding the Gluon Scene Builder Product. For windows devices, the installer for Scene Builder natively has a copy of Oracle OpenJDK within it’s install files. For example, Scene Builder 19 comes with Oracle Open JDK 19 located in the %INSTALL_LOCATION%\SceneBuilder\runtime\bin and %INSTALL_LOCATION%\SceneBuilder\runtime\lib. The Problem I am currently facing with this is that we are actively pursuing a full management over the types of Java we install on machines in our workplace including the authoritive signers of the java.

In the current situation I would like to be able to run Scene Builder 19 with another version of OpenJDK such as version 17 which we have chosen ourselves. Specifically, we are looking to swap the Oracle OpenJDK to another vendors OpenJDK such as Microsoft OpenJDK or RedHat OpenJDK. Is this possible to do?

I can see that in the “About JavaFX Scene Builder” that Scene Builder does recognize other versions of Java are available on the system (displaying the list of alternatives I had in C:\Program Files\Java\) placing them under the natively installed version. In my attempts to remove the Open JDK (runtime folder) this resulted in an error “Failed to find JVM in “C:\SceneBuilder\runtime” Directory.” 

I then alternatively tried to adjust the version by replacing the runtime folder with another OpenJDK which resolved the error, however Scene Builder would not open. My last method of attack was to adjust the “SceneBuilder.cfg” file setting the “java-options=-Djpackage.app-version” to null as I had read this stops Scene Builder from running via the native OpenJDK but this also did not work.
Is there a way to swap the version of Open JDK being used, or a way to download Gluon Scene Builder without OpenJDK as part of the install?

**Comment by Oliver-Loeffler** [comment]
Hi, most of the downloadable versions of Scene Builder have been packaged using the `jpackage` command which eventually bundles a custumized runtime with the SceneBuilder jars.

The easiest way to achieve what you have described is to clone the repository and create your own version of SceneBuilder based on a JDK of your choice. Alternatively you could try to run the jar alone.

When looking for a description how to build your own image, feel free to look into the github actions folder and the Scripts folder. The packaging is automated. 

At a later point, I could add a description how to do it to the readme.

**Reply by tmcke01** [reply]
Hi Oliver, thanks for the detailed response. 

This is exactly what I am looking to do, I will go ahead and attempt to repackage myself. Although if you had time to write a description of how to do this, it would be much appreciated as I won't say this is an area I expertise in. 

I would also be curious to test the standalone scenebuilder jar file any chance you could point me in the direction of where it is located within the repository. 

Thanks T.
